src/frontend/komponenter/Behandling/Vurdering/EndreVurdering.tsx

```diff
@@ -1,180 +1,73 @@
-        delvilkårsvurderinger: delvilkårsvurderinger,
-        resultat:
-            oppdatertDelvilkår.resultat === Vilkårsresultat.JA
-                ? Vilkårsresultat.JA
-                : vurdering.resultat,
-        unntak: undefined,
-    };
-};
-
-const hentResultatForUnntak = (unntakType: UnntakType | undefined): Vilkårsresultat => {
-    if (!unntakType) {
-        return Vilkårsresultat.IKKE_VURDERT;
-    } else if (unntakType === UnntakType.HAR_IKKE_UNNTAK) {
-        return Vilkårsresultat.NEI;
-    } else {
-        return Vilkårsresultat.JA;
-    }
-};
-
 interface Props {
     config: IVilkårConfig;
     data: IVurdering;
     oppdaterVurdering: (vurdering: IVurdering) => Promise<Ressurs<string>>;
     settRedigeringsmodus: (erRedigeringsmodus: boolean) => void;
 }
+
 const EndreVurdering: FC<Props> = ({ config, data, oppdaterVurdering, settRedigeringsmodus }) => {
     const [feilmelding, setFeilmelding] = useState<string | undefined>(undefined);
     const [oppdatererVurdering, settOppdatererVurdering] = useState<boolean>(false);
     const [vurdering, settVurdering] = useState<IVurdering>(data);
-
-    const nesteDelvilkårSomManglerVurdering = vurdering.delvilkårsvurderinger.find(
-        (delvilkår) => delvilkår.resultat === Vilkårsresultat.IKKE_VURDERT
-    );
-    let harPassertSisteDelvilkårSomSkalVises = false;
-    const sisteDelvilkår: IDelvilkår =
-        vurdering.delvilkårsvurderinger[vurdering.delvilkårsvurderinger.length - 1];
     return (
         <StyledEndreVurdering>
-            {vurdering.delvilkårsvurderinger.map((delvilkår) => {
-                if (harPassertSisteDelvilkårSomSkalVises) {
-                    return null;
-                }
-                if (
-                    nesteDelvilkårSomManglerVurdering?.type === delvilkår.type ||
-                    delvilkår.resultat === Vilkårsresultat.JA
-                ) {
-                    harPassertSisteDelvilkårSomSkalVises = true;
-                }
-                return (
-                    <RadioGruppe legend={delvilkårTypeTilTekst[delvilkår.type]}>
-                        {[Vilkårsresultat.JA, Vilkårsresultat.NEI].map((vilkårsresultat) => (
-                            <Radio
-                                key={vilkårsresultat}
-                                label={vilkårsresultatTypeTilTekst[vilkårsresultat]}
-                                name={delvilkår.type}
-                                onChange={() =>
-                                    settVurdering(
-                                        oppdaterDelvilkår(vurdering, {
-                                            type: delvilkår.type,
-                                            resultat: vilkårsresultat,
-                                        })
-                                    )
+            {config.vurdering({
+                config,
+                vurdering,
+                settVurdering,
+                lagreKnapp: (visLagreKnapp) =>
```

Comment:
Ser inte heeeelt poänget med att lagreKnapp är en funktion her

Reply:
Ser den. Er ikke helt sikker selv så gott innspill.
Jeg gjorde det slik først fordi tanken var att vi må ha flere ulike "vurderingskomponenter" og for å slippe å sende runt state mellom de løste jag det på den måten.. 

Man hadde vel egentlige ønsket att knappen var en egen komponent, men det ble fort styggt med vurdering, oppdatervurdering. Full kaos med alla sett* som skulle sendes runt :P

Reply:
Spørsmålet er om "onClick"-funksjonen skulle blitt sendt inn istedenfor selve "knapp"-komponenten? 
Og isåfall kan vi trekke ut funksjonen til en `const lagreVurdering = () => {}` og sende med  `lagreVurdering={lagreVurdering}`, kanskje?

?

Så vet "vurdering" selv om den skal vise lagreknappen?

(Hvis kommentaren bare ble vissvass 😂 kan vi ta 5 min prat om forslaget - jeg vet ikke hvor rea )

Reply:
Kan kanskje sende med `lagre: () => {lagreVurdering}` hvor `lagreVurdering` er det som skjer inni onClick?

Isåfall blir det opp til "config.vurdering" å rendre en knapp

Reply:
(Noe buggy i github - trodde den første kommentaren ikke ble lagret - dobbelposting av meg)